sdk/scala-sdk/src/main/scala/kalix/scalasdk/action/Action.scala

```diff
@@ -154,21 +154,13 @@ abstract class Action {
   @volatile
   private var _actionContext: Option[ActionContext] = None
 
+  @volatile
+  private var _timerScheduler: Option[TimerScheduler] = None
+
   /**
    * An ExecutionContext to use when composing Futures inside Actions.
-   *
-   * Note that this ExecutionContext is only available when handling a message. It will throw an exception if accessed
-   * from constructor.
    */
-  implicit lazy val executionContext: ExecutionContext = {
-    actionContext("ExecutionContext is only available when handling a message") match {
-      case ScalaActionContextAdapter(actionContext: ActionContextImpl) => actionContext.system.dispatcher
-      // should not happen as we always need to pass ScalaActionContextAdapter(ActionContextImpl)
-      case other =>
-        throw new RuntimeException(
-          s"Incompatible ActionContext instance. Found ${other.getClass}, expecting ${classOf[ActionContextImpl].getName}")
-    }
-  }
+  implicit lazy val executionContext: ExecutionContext = ExecutionContext.global
```

Comment:
I started by inject the ActorSystem, but won't work because we don't have one when using the testkit, so need to be `ExCtxt.global`. @johanandren, you will like it, no? :-)

We need the execution context because timers return Future and chances are that users will flatMap with something else. But, as explained above, that something else will probably be a call to component. 

So, using `ExecutionContext.global` kind of solve the problem, but only for a limit number of cases. 

On top of that, the idea of making a `Materializer` always available, as [discussed here](https://github.com/lightbend/kalix-jvm-sdk/pull/947#discussion_r863834641) becomes problematic because of the testkit.